workspaces/adventure-pack/goodies/typescript/Map.groupBy/index.test.ts

```diff
@@ -0,0 +1,157 @@
+import { describe, expect, it } from "@jest/globals";
+import "./index";
+
+describe("Map.groupBy", () => {
+  it("should group elements by the result of the callback function", () => {
+    const numbers = [1, 2, 3, 4, 5, 6];
+    const result = Map.groupBy(numbers, (num) =>
+      num % 2 === 0 ? "even" : "odd",
+    );
+
+    expect(result.size).toBe(2);
+    expect(result.get("even")).toStrictEqual([2, 4, 6]);
+    expect(result.get("odd")).toStrictEqual([1, 3, 5]);
+  });
+
+  it("should handle empty iterables", () => {
+    const emptyArray: number[] = [];
+    const result = Map.groupBy(emptyArray, (num) => num.toString());
+
+    expect(result.size).toBe(0);
```

Comment:
Here, it could be nice to assert that the result type is a Map.

```suggestion
    expect(result).toStrictEqual(new Map());
```